java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -1251,6 +1252,7 @@ public void setNull(int index) {
     // We need to check and reallocate the validity buffer
     while (index >= getValueCapacity()) {
       reallocValidityBuffer();
+      reallocViewBuffer();
```

Comment:
@danepitkin It's because getValueCapacity ` return Math.min(valueBufferCapacity, validityCapacity)`. If the valueBufferCapacity is lower, it will fall into a dead loop. 
And another solution is to change the while condition to `index >= getValidityBufferValueCapacity()` so that it only realloc view buffer when we add non-null values like you said. But how to defined `getValueCapacity` in this case, it will return 0.

Reply:
@danepitkin the new test cases added in this PR, checks a path the current code didn't evaluate. I proposed this patch because of the reason @ViggoC mentioned. Maybe we could do better here? 
@danepitkin @lidavidm

Reply:
I think it's a common problem. For example, BaseFixedWidthVector#setNull also reallocate validity buffer and data buffer at same time. By the way, I think we can simplify this while block to `handleSafe(index, 0)` if  we do need to realloc view buffer.

Reply:
I think it would be bad here to only reallocate the validity buffer without also expanding the view buffer, since you could reasonably conclude after setNull that there should at least be storage for the slot even if it's null.